lang/src/org/partiql/lang/eval/ExprValueFactory.kt

```diff
@@ -205,6 +220,34 @@ private class ExprValueFactoryImpl(override val ion: IonSystem) : ExprValueFacto
     override fun newTimestamp(value: Timestamp): ExprValue =
         TimestampExprValue(ion, value)
 
+    override fun newTime(value: LocalTime, zoneOffset: ZoneOffset?): ExprValue {
+        return TimeExprValue(ion, value, zoneOffset)
+    }
+
+    override fun newTime(hour: Int, minute: Int, second: Int, nano: Int, precision: Int?, tz_minutes: Int?): ExprValue {
```

Comment:
You can use default values instead of nullable parameters:

```
@JvmOverloads
override fun newTime(
    hour: Int, 
    minute: Int, 
    second: Int, 
    nano: Int, 
    precision: Int = DEFAULT_TIME_PRECISION,
    tz_minutes: Int? = null
): ExprValue { ... }
```
An exception might be if you have a parameter that legitimately needs to be `null`.  From my read `tz_minutes` does need to be nullable, so I have specified a default value of `null` and left it nullable.   When in Kotlin-land, I would definitely avoid using `null` as a way for the caller to indicate "please use the default value".    

We really should be using [`@JvmOverloads`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-overloads/)on all functions that have default parameters.  Java clients will see additional functions that call the one you wrote here, specifying the default values, like so:

```
fun newTime(
    hour: Int, 
    minute: Int, 
    second: Int, 
    nano: Int
) = newTime(hour, minute, second, nano, DEFAULT_TIME_PRECISION, tz_minutes = null)
```
and 
```newTime(
    hour: Int, 
    minute: Int, 
    second: Int, 
    nano: Int, 
    precision: Int
)  = newTime(hour, minute, second, nano, precision, tz_minutes = null)
```

Reply:
This is little tricky. Using `@JvmOverloads` is not allowed on Kotlin interfaces. The overridden methods in Kotlin cannot have default values for parameters. If we want to achieve this we probably will have to write the overloads manually. I am punting this for now until we find a better alternative. If not I will publish another PR to include the overloaded methods.